Python/sysmodule.c

```diff
@@ -209,6 +204,21 @@ sys_audit_tstate(PyThreadState *ts, const char *event,
         return 0;
     }
 
+    if (event == NULL) {
+        _PyErr_SetString(ts, PyExc_ValueError,
+                         "event argument must not be NULL");
+        return -1;
+    }
+
+    /* N format is inappropriate, because you do not know
+       whether the reference is consumed by the call.
+       Assert rather than exception for perf reasons */
+    if (argFormat != NULL && strchr(argFormat, 'N')) {
```

Comment:
The drawback of moving this check is that it is now always skipped when no hooks are registered (i.e. almost always), even in the debug build.

Reply:
Oh right. I restored the assertion, and added a second one on event, in the "should_audit() is false" code path.

I also added assertions to validate sys_audit() first argument (event).

Reply:
IMO, the problem here is that the observable behaviour should not depend on whether a hook is registered. That is, user code should not have a way to tell whether audits are on.

Would it be better to fail silently?

Reply:
It's a trade-off part of audit hooks design: https://peps.python.org/pep-0578/#performance-impact